src/fonts.php

```diff
@@ -49,35 +42,17 @@ public function is_stackable_block( $block_name ) {
 			return strpos( $block_name, 'ugb/' ) === 0;
 		}
 
-		public function gather_google_fonts( $blocks ) {
-
-			$google_fonts = array();
-			foreach ( $blocks as $block ) {
-
-				// Gather all "fontFamily" attribute values
-				if ( $this->is_stackable_block( $block['blockName'] ) && is_array( $block['attrs'] ) ) {
-					foreach ( $block['attrs'] as $attr_name => $font_name ) {
-						if ( preg_match( '/fontFamily$/i', $attr_name ) ) {
-							if ( ! self::is_web_font( $font_name ) ) {
-								continue;
-							}
-							if ( ! in_array( $font_name, $google_fonts ) ) {
-								// Allow themes to disable enqueuing fonts, helpful for custom fonts.
-								if ( apply_filters( 'stackable_enqueue_font', true, $font_name ) ) {
-									$google_fonts[] = $font_name;
-								}
-							}
-						}
-					}
-				}
+		public static function register_font( $font_name ) {
+			if ( ! self::is_web_font( $font_name ) ) {
+				return;
+			}
 
-				// Look for fonts in inner blocks.
-				if ( ! empty( $block['innerBlocks'] ) ) {
-					$google_fonts = array_unique( array_merge( $google_fonts, $this->gather_google_fonts( $block['innerBlocks'] ) ) );
+			if ( ! in_array( $font_name, self::$google_fonts ) ) {
+				// Allow themes to disable enqueuing fonts, helpful for custom fonts.
+				if ( apply_filters( 'stackable_enqueue_font', true, $font_name ) ) {
+					self::$google_fonts[] = $font_name;
```

Comment:
let's just add this if the element doesn't exist yet in the array

Reply:
already handled